services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankealueService.kt

```diff
@@ -57,7 +57,10 @@ class HankealueService(
             result.geometriat = saved?.id
         }
         result.nimi = source.nimi
-        source.haittojenhallintasuunnitelma?.let { result.haittojenhallintasuunnitelma = it }
+        source.haittojenhallintasuunnitelma?.let {
+            result.haittojenhallintasuunnitelma.clear()
+            result.haittojenhallintasuunnitelma.putAll(it)
+        }
```

Comment:
👍

Using the source map as-is could lead to trouble if the source map is modified. Even though the source map is non-mutable now, it might not always be. And I think modifying the existing maps in entities instead of re-assigning them has some benefits, though I can't remember what they were.